### PR TITLE
Set up .env.default #21

### LINKED ISSUE #21: Set up .env.default
## Tasks

- install the necessary package
- create a .env and .env.default file
- update instances where passwords were used, to now use .env

### PR DESCRIPTION
**Pull request by syedfkabir**
Resolves #21 

## Changes

- added 2 packages (`dotenv` and `dotenv-defaults`)
- updated the `knexfile` 

## Testing

Tested it by navigating the HAL browser, nothing broke!

### CODE REVIEW

**Review by ikbensiep — APPROVED** [review]
lgtm!

File: .env.defaults [hunk 1]
```diff
@@ -0,0 +1,12 @@
+PORT=8902
+MYSQL_HOST=127.0.0.1
+MYSQL_PORT=3306
+MYSQL_PASSWORD=your_password
+MYSQL_USER=tt
+MYSQL_DATABASE=tt
+PUBLIC_URI="http://localhost:8902/"
+
+AUTH_API_URI=http://localhost:8531
+OAUTH2_CLIENT_ID=
```
Comment (evert) [inline .env.defaults:10]: You can probably pick a good default for this?

File: package.json [hunk 2]
```diff
@@ -60,6 +60,8 @@
     "@curveball/problem": "^0.3.0",
     "@curveball/router": "^0.4.1",
     "@curveball/validator": "^0.9.0",
+    "dotenv": "^16.0.2",
+    "dotenv-defaults": "^5.0.2",
```
Comment (evert) [inline package.json:64]: I regret adding dotenv-defaults to our other packages. I wonder if we should just remove these?

File: src/app.ts [hunk 3]
```diff
@@ -1,3 +1,7 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+require('dotenv').config();
```
Comment (evert) [inline src/app.ts:2]: It's possible now to do this with import and not require!

File: src/knexfile.ts [hunk 4]
```diff
@@ -4,11 +4,11 @@ module.exports = {
   development: {
     client: 'mysql2',
     connection: {
-      host: '127.0.0.1',
-      port: 3306,
-      user: 'tt',
-      password: 'password',
-      database: 'tt',
+      host: process.env.MYSQL_HOST,
```
Comment (evert) [inline src/knexfile.ts:7]: Another thing that's worth testing is migrations. Do they run?

File: package.json (hunk 2 above)
Comment (syedfkabir) [reply package.json:64]: I'm not opposed to it!

File: src/knexfile.ts (hunk 4 above)
Comment (syedfkabir) [reply src/knexfile.ts:7]: they do not! hmmm

**Pull request merged**